### ISSUE TITLE
wolfi image shows critical glibc vulnerability

### ISSUE THREAD

**Issue by kapilt** [wolfi-dev/os#4125]
doing a fresh build with apko

```yaml
contents:
  packages:
    - build-base
    - python-3.11
    - git
  keyring:
    - https://packages.wolfi.dev/os/wolfi-signing.rsa.pub
  repositories:
    - https://packages.wolfi.dev/os

entrypoint:
  command: /usr/bin/python

accounts:
  groups:
    - groupname: nonroot
      gid: 65532
  users:
    - username: nonroot
      uid: 65532
      gid: 65532
  run-as: 65532
```

and then running a scan shows a critical glibc vulnerability.

![Screenshot 2023-08-01 at 6 34 42 PM](https://github.com/wolfi-dev/os/assets/21650/75eaee8a-3c6e-4827-ae28-afcb903107b0)

comparing to the chain guard python image, though which shows zero.

![Screenshot 2023-08-01 at 6 33 21 PM](https://github.com/wolfi-dev/os/assets/21650/d2af7440-c79c-4e5c-9279-e803848eb9d9)

am I missing something, cause it doesn't really add up.

**Comment by luhring** [comment]
Hi @kapilt, it looks like the new version of the glibc package (`2.38`) was published only a few hours ago, so the Python Chainguard image still has the prior version (`2.37`), and this will be true for a few more hours until the next rebuild event happens, at which point I'd expect Grype to begin surfacing CVE-2023-0687 for the Chainguard image, too.

Having said that, this CVE itself appears to be marked "disputed", and perhaps for good reason looking at [this discussion](https://sourceware.org/bugzilla/show_bug.cgi?id=29444). If we determine the CVE not to be a security issue for Wolfi, we'll update our security feed, at which point Grype will stop showing the CVE for either our image or yours.

**Comment by kaniini** [comment]
We should NAK this anyway because the "vulnerability" is actually for glibc < 2.38, e.g. we are carrying the patch for this in 2.38-r0.

I think this person was acting in bad faith when he reported this to VulDB.

**Comment by kapilt** [comment]
thanks for the details, closing this out. fwiw, going through glibc [0], the code was pretty ancient (~20yrs), its something that goes back a while on glibc versions,  re dispute status, wow glibc has pretty lengthy considerations on what they would call a security bug [1].. sadly its still going to flag on all the scanners.

[0] https://sourceware.org/git/?p=glibc.git;a=blame;f=gmon/gmon.c;hb=d165ca64980f90ccace088670652cc203d1b5411
[1] https://sourceware.org/git/?p=glibc.git;a=blob;f=SECURITY.md

**Comment by kapilt** [comment]
@kaniini good point, re the fix is included in 2.38

**Comment by kaniini** [comment]
It appears @ajayk beat me to NAKing this one already.  So it appears there is nothing remaining to do here.  In practice, this codepath would never be used in a production environment -- code built with `-pg` is just too slow for prod.
